pkg/services/ngalert/database_mig.go

```diff
@@ -68,3 +68,28 @@ func addAlertDefinitionVersionMigrations(mg *migrator.Migrator) {
 	mg.AddMigration("alter alert_definition_version table data column to mediumtext in mysql", migrator.NewRawSQLMigration("").
 		Mysql("ALTER TABLE alert_definition_version MODIFY data MEDIUMTEXT;"))
 }
+
+func alertInstanceMigration(mg *migrator.Migrator) {
+	alertInstance := migrator.Table{
+		Name: "alert_instance",
+		Columns: []*migrator.Column{
+			{Name: "definition_org_id", Type: migrator.DB_BigInt, Nullable: false},
+			{Name: "definition_uid", Type: migrator.DB_NVarchar, Length: 40, Nullable: false, Default: "0"},
+			{Name: "labels", Type: migrator.DB_Text, Nullable: false},
+			{Name: "labels_hash", Type: migrator.DB_NVarchar, Length: 190, Nullable: false},
+			{Name: "current_state", Type: migrator.DB_NVarchar, Length: 190, Nullable: false},
+			{Name: "current_state_since", Type: migrator.DB_BigInt, Nullable: false},
+			{Name: "last_eval_time", Type: migrator.DB_BigInt, Nullable: false},
+		},
+		PrimaryKeys: []string{"definition_org_id", "definition_uid", "labels_hash"},
+		Indices: []*migrator.Index{
+			{Cols: []string{"definition_org_id", "definition_uid", "current_state"}, Type: migrator.IndexType},
+			{Cols: []string{"definition_org_id", "current_state"}, Type: migrator.IndexType},
+		},
+	}
+
+	// create table
+	mg.AddMigration("create alert_instance table", migrator.NewAddTableMigration(alertInstance))
+	mg.AddMigration("add index in alert_instance table on definition_org_id, definition_uid and current_state columns", migrator.NewAddIndexMigration(alertInstance, alertInstance.Indices[0]))
```

Comment:
This causes MySQL to fail with:
`common_test.go:26: Failed to init test database: migration failed: Error 1059: Identifier name 'IDX_alert_instance_definition_org_id_definition_uid_current_state' is too long`
the index name [is derived ](https://github.com/grafana/grafana/blob/4dd7b7a82d476a7a0c29b30a8fce5b45a4c29cff/pkg/services/sqlstore/migrator/types.go#L55) from the columns of the index therefore I have to rename the columns to something shorter.